src/SearchCriteria/SearchCriteria.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+namespace Acquia\ContentHubClient\SearchCriteria;
+
+use DateTimeInterface;
+
+/**
+ * Class SearchCriteria
+ *
+ * @package Drupal\acquia_contenthub\Client
+ */
+class SearchCriteria implements \JsonSerializable
+{
+    const HEADER_NAME = 'X-Search-Criteria';
```

Comment:
"X-Search-Criteria" is quite generic. Should we prefix this? What else is in the request headers right now? Any precedence we can follow, say naming it "X-Acquia-Content-Hub-Search-Criteria"?